inc/classes/optimization/JS/class-minify.php

```diff
@@ -16,50 +16,48 @@ class Minify extends Abstract_JS_Optimization {
 	 * @since 3.1
 	 * @author Remy Perona
 	 *
+	 * @param string $html HTML content.
 	 * @return string
 	 */
-	public function optimize() {
-		$nodes = $this->find( 'script[src*=".js"]' );
+	public function optimize( $html ) {
+		$scripts = $this->find( '<script\s+([^>]+[\s\'\"])?src\s*=\s*[\'\"]\s*?([^\'\"]+\.js(?:\?[^\'\"]*)?)\s*?[\'\"]([^>]+)?\/?>', $html );
```

Comment:
No need to escape double quotes: `[\s\'\"]`, `[\'\"]`, `[^\'\"]` => `[\s"\']`, `["\']`, `[^"\']`.